subsys/bluetooth/host/classic/avctp.c

```diff
@@ -0,0 +1,151 @@
+/** @file
+ *  @brief Audio Video Control Transport Protocol
+ */
+
+/*
+ * Copyright (c) 2015-2016 Intel Corporation
+ * Copyright (C) 2024 Xiaomi Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <strings.h>
+#include <errno.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/util.h>
+
+#include <zephyr/bluetooth/hci.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/l2cap.h>
+#include <zephyr/bluetooth/classic/avctp.h>
+
+#include "avctp_internal.h"
+#include "host/hci_core.h"
+#include "host/conn_internal.h"
+#include "l2cap_br_internal.h"
+
+#define LOG_LEVEL CONFIG_BT_AVCTP_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(bt_avctp);
+
+#define AVCTP_CHAN(_ch) CONTAINER_OF(_ch, struct bt_avctp, br_chan.chan)
+
+K_MUTEX_DEFINE(avctp_mutex);
+#define AVCTP_LOCK() k_mutex_lock(&avctp_mutex, K_FOREVER)
+#define AVCTP_UNLOCK() k_mutex_unlock(&avctp_mutex)
+
+static struct bt_avctp_event_cb *event_cb;
+
+static void avctp_l2cap_connected(struct bt_l2cap_chan *chan)
+{
+	struct bt_avctp *session;
+
+	if (!chan) {
+		LOG_ERR("Invalid AVCTP chan");
+		return;
+	}
+
+	session = AVCTP_CHAN(chan);
+	LOG_DBG("chan %p session %p", chan, session);
+
+	session->ops->connected(session);
+}
+
+static void avctp_l2cap_disconnected(struct bt_l2cap_chan *chan)
+{
+	struct bt_avctp *session = AVCTP_CHAN(chan);
+
+	LOG_DBG("chan %p session %p", chan, session);
+	session->br_chan.chan.conn = NULL;
+
+	session->ops->disconnected(session);
```

Comment:
Same here, `session->ops` and `session->ops->disconnected` need to check before using it.